### PR TITLE
forge: add support for fullnodes

### PR DESCRIPTION
**Pull request by bmwill**

### CODE REVIEW

File: testsuite/forge/src/backend/local/swarm.rs [hunk 1]
```diff
@@ -310,39 +345,43 @@ impl Swarm for LocalSwarm {
...
         todo!()
     }
 
     fn remove_validator(&mut self, _id: PeerId) -> Result<()> {
         todo!()
     }
 
-    fn add_full_node(&mut self, _id: PeerId) -> Result<()> {
-        todo!()
+    fn add_full_node(&mut self, version: &Version, template: NodeConfig) -> Result<PeerId> {
```
Comment (gregnazario) [inline testsuite/forge/src/backend/local/swarm.rs:375]: You know we should really decide someday on the standard of `fullnode` vs `full_node`

File: config/management/genesis/src/validator_builder.rs [hunk 2]
```diff
@@ -305,19 +307,53 @@ impl ValidatorBuilder {
...
+            listen_address: fullnode_network_listen_address,
+            network_id: NetworkId::Public,
+            max_outbound_connections: 0,
+            identity: Identity::from_storage(
+                FULLNODE_NETWORK_KEY.to_owned(),
+                OWNER_ACCOUNT.to_owned(),
+                SecureBackend::OnDiskStorage(validator.storage_config.clone()),
+            ),
+
+            ..Default::default()
```
Comment (gregnazario) [inline config/management/genesis/src/validator_builder.rs:329]: Does `..` by itself not do default?

**Review by gregnazario — DISMISSED** [review]

File: config/management/genesis/src/validator_builder.rs (hunk 2 above)
Comment (bmwill) [reply config/management/genesis/src/validator_builder.rs:329]: no, it takes an arbitrary struct as input.

File: testsuite/forge/src/backend/local/swarm.rs (hunk 1 above)
Comment (bmwill) [reply testsuite/forge/src/backend/local/swarm.rs:375]: yeah seriously, even i do it inconsistently

**Pull request merged**